packages/vhd-lib/createStreamNbd.js

```diff
@@ -132,6 +132,12 @@ exports.createNbdVhdStream = async function createVhdStream(
       }
     }
 
+    sourceStream.on('error', error => {
+      /* we can safely ignore the UND_ERR_ABORTED errror since we are the one aborting it */
+      if (error.code !== 'UND_ERR_ABORTED') {
+        throw error
+      }
+    })
```

Comment:
In this case, I think you can simply ignore the error `.on('error', noop)` as we are destroying the error right after adding this listener.

No need to couple this with `undici` as adding this listener will work well no matter where the stream comes from.

Reply:
You can even write this as `sourceStream.on('error', noop).destroy()`.